x-pack/plugins/dashboard_enhanced/public/services/drilldowns/dashboard_to_dashboard_drilldown/components/collect_config_container.tsx

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { EuiComboBoxOptionOption } from '@elastic/eui';
+import { debounce, findIndex } from 'lodash';
+import { SimpleSavedObject } from '../../../../../../../../src/core/public';
+import { DashboardDrilldownConfig } from './dashboard_drilldown_config';
+import { txtDestinationDashboardNotFound } from './i18n';
+import { CollectConfigProps } from '../../../../../../../../src/plugins/kibana_utils/public';
+import { UiActionsEnhancedDrilldownActionFactoryContext as DrilldownFactoryContext } from '../../../../../../advanced_ui_actions/public';
+import { Config } from '../types';
+import { IEmbeddable } from '../../../../../../../../src/plugins/embeddable/public';
+import { Params } from '../drilldown';
+
+const mergeDashboards = (
+  dashboards: Array<EuiComboBoxOptionOption<string>>,
+  selectedDashboard?: EuiComboBoxOptionOption<string>
+) => {
+  // if we have a selected dashboard and its not in the list, append it
+  if (selectedDashboard && findIndex(dashboards, { value: selectedDashboard.value }) === -1) {
+    return [selectedDashboard, ...dashboards];
+  }
+  return dashboards;
+};
+
+const dashboardSavedObjectToMenuItem = (
+  savedObject: SimpleSavedObject<{
+    title: string;
+  }>
+) => ({
+  value: savedObject.id,
+  label: savedObject.attributes.title,
+});
+
+interface DashboardDrilldownCollectConfigProps extends CollectConfigProps<Config> {
+  params: Params;
+  context: DrilldownFactoryContext<{
+    embeddable: IEmbeddable;
+  }>;
+}
```

Comment:
```suggestion
interface DashboardDrilldownCollectConfigProps extends CollectConfigProps<Config, DrilldownFactoryContext<{
    embeddable: IEmbeddable;
  }>> {
  params: Params;
}
```

Maybe that instead ^^?

Reply:
Yes, it should be 

```ts
interface DashboardDrilldownCollectConfigProps
  extends CollectConfigProps<Config, DrilldownFactoryContext<EmbeddableContext>> {
  params: Params;
}
```

Done.